portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -11,22 +11,94 @@
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.sps.data.Message;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns comments */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public final class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query("Message").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List < Message > allMessages = new ArrayList < > ();
+    for (Entity entity: results.asIterable()) {
+      long id = entity.getKey().getId();
+      String comment = (String) entity.getProperty("comment");
+      Boolean yes = (Boolean) entity.getProperty("need response");
+      String email = (String) entity.getProperty("email");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Message curMessage = new Message(id, comment, yes, email, timestamp);
+      allMessages.add(curMessage);
+    }
+    // Send the JSON as the response
+    response.setContentType("application/json");
+    Gson gson = new Gson();
+    response.getWriter().println(gson.toJson(allMessages));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String msg = request.getParameter("text-input");
+    Boolean yes = getYes(request);
+    String email = request.getParameter("email-input");
+    // Record time submitted.
+    long timestamp = System.currentTimeMillis();
+
+    response.setContentType("text/html");
+    response.getWriter().println("Thanks for reaching out!");
+
+    if (yes) {
+      response.setContentType("text/html");
+      response.getWriter().println("I'll be in touch soon!");
+    }
+
+    // creates Entity for messages
+    Entity messageEntity = new Entity("Message");
+    messageEntity.setProperty("comment", msg);
+    messageEntity.setProperty("need response", yes);
+    messageEntity.setProperty("email", email);
+    messageEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(messageEntity);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  /*
+  Function to check if email is submitted if the user wants a response. Don't think this function actually does anything.
```

Comment:
"Don't think this function actually does anything." 🤔  So should this be removed or fixed?